bindings_node/src/mls_client.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::HashMap;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use crate::conversations::NapiConversations;
+use napi::bindgen_prelude::{Error, Result, Uint8Array};
+use napi_derive::napi;
+use xmtp_api_grpc::grpc_api_helper::Client as TonicApiClient;
+use xmtp_cryptography::signature::ed25519_public_key_to_address;
+use xmtp_id::associations::{Erc1271Signature, RecoverableEcdsaSignature};
+use xmtp_mls::builder::ClientBuilder;
+use xmtp_mls::identity::IdentityStrategy;
+use xmtp_mls::storage::{EncryptedMessageStore, EncryptionKey, StorageOption};
+use xmtp_mls::Client as MlsClient;
+
+pub type RustXmtpClient = MlsClient<TonicApiClient>;
+
+#[napi]
+pub struct NapiClient {
+  inner_client: Arc<RustXmtpClient>,
+  pub account_address: String,
+}
+
+#[napi]
+pub async fn create_client(
+  host: String,
+  is_secure: bool,
+  db_path: String,
+  account_address: String,
+  encryption_key: Option<Uint8Array>,
+) -> Result<NapiClient> {
+  let api_client = TonicApiClient::create(host.clone(), is_secure)
+    .await
+    .map_err(|_| Error::from_reason("Error creating Tonic API client"))?;
+
+  let storage_option = StorageOption::Persistent(db_path);
+
+  let store = match encryption_key {
+    Some(key) => {
+      let key: Vec<u8> = key.deref().into();
+      let key: EncryptionKey = key
+        .try_into()
+        .map_err(|_| Error::from_reason("Malformed 32 byte encryption key".to_string()))?;
+      EncryptedMessageStore::new(storage_option, key)
+        .map_err(|_| Error::from_reason("Error creating encrypted message store"))?
+    }
+    None => EncryptedMessageStore::new_unencrypted(storage_option)
+      .map_err(|_| Error::from_reason("Error creating unencrypted message store"))?,
+  };
+
+  let identity_strategy =
+    IdentityStrategy::CreateIfNotFound(account_address.clone().to_lowercase(), None);
+
+  let xmtp_client = ClientBuilder::new(identity_strategy)
+    .api_client(api_client)
+    .store(store)
+    .build()
+    .await
+    .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+  Ok(NapiClient {
+    inner_client: Arc::new(xmtp_client),
+    account_address,
+  })
+}
+
+#[napi]
+impl NapiClient {
+  #[napi]
+  pub fn inbox_id(&self) -> String {
+    self.inner_client.inbox_id()
+  }
+
+  #[napi]
+  pub fn is_registered(&self) -> bool {
+    self.inner_client.identity().signature_request().is_none()
+  }
+
+  #[napi]
+  pub fn installation_id(&self) -> String {
+    ed25519_public_key_to_address(self.inner_client.installation_public_key().as_slice())
+  }
+
+  #[napi]
+  pub async fn can_message(&self, account_addresses: Vec<String>) -> Result<HashMap<String, bool>> {
+    let results: HashMap<String, bool> = self
+      .inner_client
+      .can_message(account_addresses)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    Ok(results)
+  }
+
+  #[napi]
+  pub async fn register_ecdsa_identity(&self, signature_bytes: Uint8Array) -> Result<()> {
```

Comment:
This is fine now, but if you register with legacy key, you won't need to provide `siganture_bytes`